FormSpec: Optimize formspec form size

v-rob opened the pull request:
This PR optimizes the formspec form size to use the screen space more efficiently and generally makes it more usable.  It does the following:
* Makes `real_coordinates` formspecs use more of the available screen space instead of factoring in the unused spacing and padding.
* Makes all formspecs stay within the screen boundaries (so this PR closes #7603), and also makes a 5% of the screen size spacing on all sides to make sure the formspec doesn't touch the screen edges.
* Removes legacy Android-specific code that uses the wrong numbers and doesn't do what it says.

## To do

This PR is Ready for Review.  Very simple.

## How to test

Try making formspecs of varying sizes in real and old coordinates and ensure that they stay inside the screen but also take full advantage of the space available.

SmallJoker commented on src/gui/guiFormSpecMenu.cpp:
```diff
@@ -3098,22 +3098,36 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
+					((5.0 / 4.0) * (0.5 + mydata.invsize.X));
+				double fity_imgsize = mydata.screensize.Y /
+					((15.0 / 13.0) * (0.85 * mydata.invsize.Y));
+				double screen_dpi = RenderingEngine::getDisplayDensity() * 96;
```
This variable already exists on L3069.

v-rob replied:
I wonder why GCC didn't give me an unused variable warning since it's no longer used...

v-rob commented:
Added a 5% border on all sides, and it does look pretty good.  Actually, formspecs were still the wrong size in the Y direction using the old coordinate system; there was a multiplication where there should have been an addition (ever since 0.4.something; this is probably the main reason for #7603).  This is why I took me so long to respond; I couldn't figure out what was the problem :)

I also removed the weird and incorrect Android code which uses the wrong numbers and doesn't even do what the comment says, so Android uses the same code as desktop, except it retains the preferred imgsize.  At least the Android code had the addition instead of multiplication...

SmallJoker reviewed (approved):
Works.

paramat commented:
v-rob, good to see you are now a core dev.

2 core dev approvals are required to merge a PR.
If a core dev is the author of a PR, their approval of their own PR counts as 1 of the required 2 core dev approvals. This is automatically assumed, so adding the '>= Two approvals' label is not necessary, the label can be left as 'One approval'.
